PlasterTemplate/PlasterTemplates/GitLab/psake.ps1

```diff
@@ -268,16 +273,32 @@ Task Deploy -Depends TestAfterBuild {
     "`nInvoking PSDeploy"
     Invoke-PSDeploy @Verbose @Params
 
-    <#
-    "`tSetting git repository url"
-    $GitLabUrl = 'https://{0}@gitlab.com/<%= $PLASTER_PARAM_GitLabUserName %>/{1}.git' -f $ENV:GitLab_PAT, $env:BHProjectName
-
-    "`tPushing built docs to GitLab"
-    git add "$env:BHProjectPath\docs\*"
-    git add "$env:BHModulePath\CHANGELOG.md"
-    git commit -m "Bump version to $ReleaseVersion`n[ci skip]"
-    # --porcelain is to stop git sending output to stderr
-    git push $GitLabUrl HEAD:master --porcelain
-    "`n"
-    #>
+<%
+    if ($PLASTER_PARAM_DeployDocs -eq 'Yes') {
+        '   "`tSetting git repository url"'
+        '   # This may need to be changed if your gitlab server does not use HTTPS'
+        $GitLabURL = [uri]$PLASTER_PARAM_GitLabURL
+        if ($GitLabURL.Scheme) {
+        $GitLabProtocol = $GitLabURL.Scheme
+        }
+        else {
+            $GitLabProtocol = https
+        }
+        if ($GitLabURL.Host) {
+            $GitLabHost = $GitLabURL.Host
+        }
+        else {
+            $GitLabHost = $GitLabURL.OriginalString
+        }
+        "    `$GitLabUrl = '$($GitLabProtocol)://oauth2:{0}@$($GitLabHost)/$($PLASTER_PARAM_GitLabUserName)/{1}.git' -f `$env:GitLab_PAT, `$env:BHProjectName"
+        '   [version]$ReleaseVersion = git describe --tags'
+        '   "`tPushing built docs to GitLab"'
+        '   git add "$env:BHProjectPath\docs\*"'
+        '   git add "$env:BHProjectPath\CHANGELOG.md"'
+        '   git commit -m "Bump version to $ReleaseVersion`n[ci skip]"'
+        '   # --porcelain is to stop git sending output to stderr'
+        '   git push $GitLabUrl HEAD:master --porcelain'
+        '   "`n"'
```

Comment:
These need one more leading space inside the quotes as it's only 3 at the moment